src/core/src/UsdUtil.cpp

```diff
@@ -246,7 +246,7 @@ bool primExists(const pxr::UsdStageWeakPtr& pStage, const pxr::SdfPath& path) {
 }
 
 bool isSchemaValid(const pxr::UsdSchemaBase& schema) {
-    return schema.GetSchemaKind() != pxr::UsdSchemaKind::Invalid;
+    return schema.GetPrim().IsValid() && schema.GetSchemaKind() != pxr::UsdSchemaKind::Invalid;
```

Comment:
I can see this working, but I'm wondering if it changes the intent of the method. Most of the calls are `!UsdUtil::isSchemaValid(cesiumTileset)`, where we seem to be checking for prim existence. Why are we checking for schema validity instead of prim existence?

Reply:
I think it's worth checking both in case the schema is created from a prim with the wrong type. I'm not sure when that would actually happen, but the extra safeguard doesn't hurt.

Reply:
True. I think a better way to ask is "What do we get from checking schema validity vs checking for prim existence across the code base?" Could it all just be a call to [Usd.Object.IsValid()](https://docs.omniverse.nvidia.com/dev-guide/latest/programmer_ref/usd/prims/check-prim-exists.html)? `!UsdUtil::isSchemaValid(cesiumTileset)` makes me wonder if there's something particularly special about the schema that needs to be checked. Or is `isSchemaValid` a more robust version of `primExists`?

Reply:
> Or is isSchemaValid a more robust version of primExists?

That's how I see it, since a schema wraps a prim.